apps/zui/src/domain/loads/temp-file-holder.ts

```diff
@@ -0,0 +1,41 @@
+import os from "os"
+import path from "path"
+import * as fs from "fs-extra"
+import {getUniqName} from "src/util/get-uniq-name"
+
+export class TempFileHolder {
+  dir: string
+
+  constructor(namespace: string) {
+    this.dir = path.join(os.tmpdir(), namespace)
+    fs.ensureDirSync(this.dir)
```

Comment:
I can see that this is now creating the multiple paste files in a temporary directory whose name is derived from the username, e.g., `phil_pastes` for me. Revisiting my PR comment about concern that in a Linux style environment there might be collision among the temp files of multiple users, it seems that (at least in theory) that collision exposure now exists among these temp directories. Since it looks like we're leveraging the user's OS username I guess a collision is nearly impossible since OS usernames are assumed to be unique. If it's not difficult to do, I'd feel less paranoid if we were using `fs.mkdtemp()` in some way since that adds additional uniqueness, e.g. like in this sample program:

```
const fs = require('fs'); 
  
fs.mkdtemp("phil_pastes-", (err, folder) => { 
  if (err) 
    console.log(err); 
  else { 
    console.log("The temporary folder path is:", folder); 
  } 
});
```